pkg/driver/controllerserver.go

```diff
@@ -162,13 +234,54 @@ func (cs *controllerServer) DeleteVolume(_ context.Context, req *csi.DeleteVolum
 	klog.Infof("Deleting volume %v", volumeID)
 	secretMap := req.GetSecrets()
 
-	creds, err := getCredentials(req.GetSecrets())
+	var endPoint, locationConstraint string
+
+	if len(secretMap) == 0 {
+		klog.Info("Did not find the secret that matches pvc name. Fetching custom secret from PVC annotations")
+
+		pv, err := utils.GetPV(volumeID)
+		if err != nil {
+			return nil, err
+		}
+
+		klog.Info("pv Resource details:\n\t", pv)
+
+		secretName := pv.Spec.CSI.NodePublishSecretRef.Name
+		secretNamespace := pv.Spec.CSI.NodePublishSecretRef.Namespace
+
+		if secretName == "" {
+			return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("Secret details not found, could not fetch the secret %v", err))
+		}
+
+		if secretNamespace == "" {
+			klog.Info("secret Namespace not found. trying to fetch the secret in default namespace")
+			secretNamespace = constants.DefaultNamespace
+		}
+
+		endPoint = pv.Spec.CSI.VolumeAttributes["cosEndpoint"]
+		locationConstraint = pv.Spec.CSI.VolumeAttributes["locationConstraint"]
+
+		klog.Info("secret details found. secret-name: ", secretName, "\tsecret-namespace: ", secretNamespace)
+
+		secret, err := utils.GetSecret(secretName, secretNamespace)
+		if err != nil {
+			return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("Secret resource not found %v", err))
+		}
+
+		secretMapCustom, err := parseCustomSecret(secret)
+		if err != nil {
+			return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("Error in reading secret parameters %v", err))
+		}
+
+		klog.Info("custom secret parameters parsed successfully")
+		secretMap = secretMapCustom
+	}
+
+	creds, err := getCredentials(secretMap)
 	if err != nil {
-		return nil, fmt.Errorf("cannot get credentials: %v", err)
+		return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("Error in getting credentials %v", err))
 	}
 
-	endPoint := secretMap["cosEndpoint"]
-	locationConstraint := secretMap["locationConstraint"]
```

Comment:
I guess this we still need in else part
As per current code,  if len(secretMap) != 0 then endPoint & locationConstraint will be empty strings

Reply:
Added in the beginning of function